locustempus/main/permissions.py

```diff
@@ -20,3 +21,71 @@ def has_object_permission(self, request, view, obj):
             return True
 
         return obj.course.is_faculty(request.user)
+
+
+class IsResponseOwnerOrFaculty(permissions.IsAuthenticated):
+    """
+    This permission class implements the following:
+
+    List GET request, w/ no querystring args:
+    - Only authed users, can only retrieve responses that they own
+
+    List GET w/ activity querystring
+    - If user is faculty for the course for that activity, then return
+      all response objects that have that activity as a foriegn key
+    _ If a user is a student in the related course, then return only
+      the responses they own.
+
+    Single GET
+    - The user must be faculty in the course
+    - OR the user must be a member of the course and an owner of the
+      response
+
+    POST, PUT, DELETE requires an activity PK
+    - If user is faculty in the related course, they don't have
+      permission to create new response for that activity
+    - If a user is student in the related course, they may create
+      a new response object
+
+    PATCH
+    - If a user is faculty for the related course, they do have permission, so
+      that they can provide a response to the reflection
```

Comment:
At the interface level, yes; at the API level no.

Reply:
I've removed the ability for faculty to edit a response object. In a later PR I'll create a Feedback model for faculty. On this new model's save method, I'll have it update the response status. This will keep the permissions clear from an API perspective.